src/Handler/IteratorHandler.php

```diff
@@ -0,0 +1,113 @@
+<?php
+
+declare(strict_types=1);
+
+namespace JMS\Serializer\Handler;
+
+use JMS\Serializer\DeserializationContext;
+use JMS\Serializer\GraphNavigatorInterface;
+use JMS\Serializer\SerializationContext;
+use JMS\Serializer\Visitor\DeserializationVisitorInterface;
+use JMS\Serializer\Visitor\SerializationVisitorInterface;
+
+final class IteratorHandler implements SubscribingHandlerInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public static function getSubscribingMethods()
+    {
+        $methods = [];
+        $formats = ['json', 'xml', 'yml'];
+        $iteratorTypes = [
+            \Iterator::class,
```

Comment:
Thanks for comment. 
During the implementation it was a nice idea for me to support the interface, without any concrete implementation. But today I'm not super sure about - in fact it may create some issues for users, when they would expect other result than me. I've decided to remove it cause it will be easier to add it than remove it in the future.